lottie-swift/src/Public/Animation/AnimationPublic.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  AnimationPublic.swift
+//  lottie-swift
+//
+//  Created by Brandon Withrow on 2/5/19.
+//
+
+import Foundation
+import CoreGraphics
+
+public extension Animation {
+  
+  // MARK: Animation (Loading)
+  
+  /**
+   Loads an animation model from a bundle by it's name. Returns `nil` if an animation is not found.
+   
+   - Parameter name: The name of the json file without the json extension. EG "StarAnimation"
+   - Parameter bundle: The bundle in which the animation is located. Defaults to `Bundle.main`
+   - Parameter subdirectory: A subdirectory in the bundle in which the animation is located. Optional.
+   - Parameter animationCache: A cache for holding loaded animations. Optional.
+   
+   - Returns: Deseralized `Animation`. Optional.
+   */
+  public static func named(_ name: String,
+                           bundle: Bundle = Bundle.main,
+                           subdirectory: String? = nil,
+                           animationCache: AnimationCacheProvider? = nil) -> Animation? {
+    /// Create a cache key for the animation.
+    let cacheKey = bundle.bundlePath + (subdirectory ?? "") + "/" + name
+    
+    /// Check cache for animation
+    if let animationCache = animationCache,
+      let animation = animationCache.animation(forKey: cacheKey) {
+      /// If found, return the animation.
+      return animation
+    }
+    /// Make sure the bundle has a file at the path provided.
+    guard let url = bundle.url(forResource: name, withExtension: "json", subdirectory: subdirectory) else {
+      return nil
+    }
+    
+    do {
+      /// Decode animation.
+      let json = try Data(contentsOf: url)
+      let animation = try JSONDecoder().decode(Animation.self, from: json)
+      animationCache?.setAnimation(animation, forKey: cacheKey)
+      return animation
+    } catch {
+      /// Decoding error.
+      return nil
+    }
+  }
+  
+  /**
+   Loads an animation from a specific filepath.
+   - Parameter filepath: The absolute filepath of the animation to load. EG "/User/Me/starAnimation.json"
+   - Parameter animationCache: A cache for holding loaded animations. Optional.
+   
+   - Returns: Deseralized `Animation`. Optional.
+   */
+  public static func filepath(_ filepath: String,
+                              animationCache: AnimationCacheProvider? = nil) -> Animation? {
+    
+    /// Check cache for animation
+    if let animationCache = animationCache,
+      let animation = animationCache.animation(forKey: filepath) {
+      return animation
+    }
+    
+    /// Make sure the file exists.
+    guard FileManager.default.fileExists(atPath: filepath) else {
```

Comment:
Using `fileExists` is against Apple recommendations. Apple advices to directly attempt the file operation without it. See the note of https://developer.apple.com/documentation/foundation/filemanager/1415645-fileexists

Reply:
👍